golem-service-base/src/storage/blob/mod.rs

```diff
@@ -328,6 +328,7 @@ impl<'a, S: BlobStorage + ?Sized + Sync> LabelledBlobStorage<'a, S> {
 #[derive(Debug, Clone, PartialEq, Eq, Hash)]
 pub enum BlobStorageNamespace {
     CompilationCache,
+    InitialComponentFiles,
```

Comment:
Let's add `AccountId` to this namespace too (and use it in the path just like with the oplog related blob stores). This makes it easier to collect / delete a specific account's data.

Reply:
Done. Note that this will increase storage use potentially as we won't be able to share data between accounts.

Reply:
Yeah it is a bit tricky because there are no accounts in the OSS part of Golem. But if you check worker executor, it propagates an account id everywhere, and various upper layers are trying to abstract it. (It's a bit of a mess right now). 

For example https://github.com/golemcloud/golem/blob/main/golem-worker-service/src/lib.rs#L11 and in the component service it's represented by the "namespace" (but the `DefaultNamespace` here https://github.com/golemcloud/golem/blob/main/golem-service-base/src/auth.rs does not contain anything, as there are no accounts in the open-source version).

As you probably already figured out, the reason for all these abstractions, and different -base crates etc is that we have some advanced features built on top of this codebase which require these features. 

Of course it's hard to figure out the best way to do this as you don't see the other half. 
For this PR is suggest the following:

- add `AccountId` to this namespace, because it already has it for `OplogPayload`, for example 
- in worker executor you already have account id available everywhere (sometimes hidden in an `OwnedWorkerId`)
- in component service, you need to extract it from the namspace which is propagated to the service level. It does not have a way to extract it right now, but you can define a trait with an `Self -> Option<AccountId>` extractor and require it for the namespace here: https://github.com/golemcloud/golem/blob/main/golem-component-service-base/src/service/component.rs#L390
- if there is no account id, use "-1" (because worker service does it already in one of the above links)

Reply:
Done. Please take a look whether this what you had in mind:
https://github.com/golemcloud/golem/pull/1032/commits/81155567c18f945c8c025e47d7edb09790f3e719

Reply:
I'm surprised it "leaked" into so many places like API definition models. Is that necessary? Worker service was already able to address workers somehow, which means it had the mechanism to get the necessary context (accountid etc) alredy without these changes.

I would like to revisit this and if possible make less spread out change.

Other than this from my side the PR is ready to go.

Reply:
The main reason this leaked into so many places is that I need to use the correct accountId for an api deployment if I understand the code correctly.

The way it was done before was that namespace information was included when creating the ApiDefinition, but it was serialized to a string and never deserialized again. Deserializing and carrying the information around as a proper type makes it show up in a lot of places https://github.com/golemcloud/golem/blob/b861a50c1a9bcab8a45eec4ad70acb9ac59802e1/golem-worker-service-base/src/repo/api_definition.rs#L27

Let me try to limit it to only the compiled versions of the definitions.

Reply:
@vigoo Done. I think this looks much better https://github.com/golemcloud/golem/pull/1032/commits/85736967003e06fe3719c2794b712a4f6fa3ad34.